Добавлено задание на учебную практику 2025.

LeahLeah88 opened the pull request:
Добавлено задание на учебную практику 2025. Исправлены замечания с нарушением инкапсуляции, добавлено сообщение об ошибке, добавлен метод для удаления последнего элемента, добавлен заголовочный файл.

LeahLeah88 committed:
Add project files.

Add project files.

tak333 commented on Praktika.cpp:
```diff
@@ -0,0 +1,184 @@
+        return size;
+    }
+
+    Node<T>* getHead() const {
```
Не давайте пользователю узлы, это нарушает инкапсуляцию и он может натворить ерунды: например, очистить память из-под узла, перенаправить его и пр.

tak333 commented on Praktika.cpp:
```diff
@@ -0,0 +1,184 @@
+
+    void remove() override {
+        if (!list.getSize())
+            return;
```
Не игнорируйте ошибочную ситуацию, сообщите о ней пользователю.

tak333 commented on Praktika.cpp:
```diff
@@ -0,0 +1,184 @@
+        if (!list.getSize())
+            return;
+        // Удаляем последний добавленный элемент (стек LIFO)
+        Node<T>* current = list.getHead();
```
Зачем идти от начала к концу, если можно сразу удалить последний (ведь в списке он хранится, напишите там соответствующий метод).

tak333 reviewed (changes requested):
Исправьте замечания + разнесите код по файлам h и cpp

LeahLeah88 committed:
Исправлены предыдущие замечания.

Исправлены замечания с нарушением инкапсуляции, добавлено сообщение об ошибке, добавлен метод для удаления последнего элемента

LeahLeah88 committed:
Добавлен заголовочный файл.

Добавлен заголовочный файл.

tak333 commented on Praktika.cpp:
```diff
@@ -0,0 +1,184 @@
+        ++size;
+    }
+
+    void remove(Node<T>* node) {
```
Откуда пользователь знает про узлы?

LeahLeah88 replied:
Добрый день!
Скажите, пожалуйста, чтобы сдать задание без правок на "удовлетворительно", нужно заново запрос на слияние отправлять?

LeahLeah88 replied:
> Откуда пользователь знает про узлы?

Добрый день! Предыдущие исправления, получается, не засчитаны? Какие ещё нужно внести?

tak333 reviewed (changes requested):
Не вижу исправления замечаний. В данном виде приму на "удовлетворительно". Иначе вносите корректировки.

tak333 commented:
Разберитесь с файлами в проекте.
Каждый класс = отдельный модуль h + cpp.
main отдельно.

LeahLeah88 commented:
> Не вижу исправления замечаний. В данном виде приму на "удовлетворительно". Иначе вносите корректировки.

Добрый день!
Скажите, пожалуйста, чтобы сдать задание без правок на "удовлетворительно", нужно заново запрос на слияние отправлять?

tak333 commented:
Добрый день!

Да, учитываете замечания: вносите исправления или задаёте вопросы, потом обновляете код и отправляйте повторно запрос на проверку.

LeahLeah88 commented:
> Добрый день!
> 
> Да, учитываете замечания: вносите исправления или задаёте вопросы, потом обновляете код и отправляйте повторно запрос на проверку.

Предыдущие исправления, получается, не засчитаны? Какие ещё нужно внести?

tak333 commented:
Прошу прощения, не так прочитала ваш вопрос сначала. Если вас устраивает "удовлетворительно", то делайте отчёт и кидайте на почту.
Если хотите повысить, то надо учесть предыдущие комментарии, а потом отправить повторно на проверку.
Стало понятней?
